sources/editor/Stride.Core.Assets.Editor/ViewModel/DirectoryBaseViewModel.cs

```diff
@@ -165,8 +165,7 @@ public DirectoryBaseViewModel GetOrCreateDirectory(string path, bool canUndoRedo
         /// <param name="newParent">The nwe parent of this directory.</param>
         protected void SetParent(DirectoryBaseViewModel oldParent, DirectoryBaseViewModel newParent)
         {
-            var directory = this as DirectoryViewModel;
-            if (directory == null) throw new InvalidOperationException("Can't change the parent of this folder");
+            if (this is not DirectoryViewModel directory) throw new InvalidOperationException("Can't change the parent of this folder");
```

Comment:
Since the slightly cleaner form of type identification is used elsewhere in this class, updating this method to be consistent with that pattern.